packages/backend-api/src/api/publisher/commentActions.ts

```diff
@@ -20,34 +20,41 @@ import {
   IKnownTasks,
 } from '@conversationai/moderator-backend-queue';
 import * as express from 'express';
-import * as Joi from 'joi';
+import { commentActionSchema } from '../services/commentActions';
 import { dataSchema, validateRequest } from '../util/validation';
 
 export const STATUS_ACCEPTED = 'accepted';
 export const STATUS_REJECTED = 'rejected';
 
-const validateNumberSchema = validateRequest(dataSchema(Joi.number()));
+const validateCommentActionRequest = validateRequest(dataSchema(commentActionSchema));
 
 /**
  * Queues an approval or rejected action. Accepts array of comment ids, or a single comment id.
  */
 export function queueMainAction(name: IKnownTasks): express.RequestHandler {
-  return async (req, res, next) => {
-    const dataArray = Array.isArray(req.body.data) ? req.body.data : [req.body.data];
-
-    for (const commentId of dataArray) {
-      const parsedCommentId = parseInt(commentId, 10);
-
-      await enqueue<ICommentActionData>(name, {
-        commentId: parsedCommentId,
-        userId: req.user ? req.user.get('id') : null,
-        isBatchAction: false,
-        autoConfirmDecision: true,
-      }, req.body.runImmediately || false);
+  return async ({ body }, res, next) => {
+    try {
+      const dataArray = Array.isArray(body.data) ? body.data : [body.data];
+
+      for (const data of dataArray) {
+        const { userId, commentId } = data;
+
+        const parsedUserId = parseInt(userId, 10);
+        const parsedCommentId = parseInt(commentId, 10);
+
+        const isBatchAction = (dataArray.length > 1) ? true : false;
+        await enqueue<ICommentActionData>(name, {
+          commentId: parsedCommentId,
+          userId: parsedUserId,
+          isBatchAction,
+        }, body.runImmediately || false);
+      }
+
+      res.json({ status: 'success' });
+      next();
+    } catch (e) {
+      console.log(e)
```

Comment:
Probably wants to be a console.error?  Probably you want to use the logging code...

```
import { logger } from '@conversationai/moderator-backend-core';

...

logger.error(e);
```